hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaCache.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.avro;
+
+import org.apache.avro.Schema;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Cache avro schema variables.
+ * NOTE: All methods that create/get Schema instances should go through this class for caching.
+ */
+public class AvroSchemaCache {
+
+
+  // Ensure that there is only one variable instance of the same schema within a entire JVM lifetime
+  private static final ConcurrentHashMap<Schema, Schema> SCHEMA_CACHE = new ConcurrentHashMap<>();
```

Comment:
Maybe guava `Interner` is a better choice for the cache.

Reply:
> Maybe guava `Interner` is a better choice for the cache.

Nice suggestion! Picked it.